ui/appui-react/src/appui-react/statusfields/SnapMode.tsx

```diff
@@ -134,6 +152,21 @@ export function SnapModeField(props: SnapModeFieldProps) {
       iconName: "snaps-bisector",
     },
   ];
+
+  if (snapModeChoices.length > 0) {
+    // Remove all snap mode not present in snapModeChoices.
+    snapModes = snapModes.filter((snapModeFieldEntry) =>
+      snapModeChoices.includes(snapModeFieldEntry.value)
+    );
+    if (
+      (!props.snapMode || !snapModeChoices.includes(props.snapMode)) &&
+      !snapModeChoices.includes(SnapMode.NearestKeypoint)
+    ) {
+      // The default snap mode have been filtered out. Default to the first one.
+      snapMode = snapModeChoices[0];
+    }
+  }
+
```

Comment:
Can we refactor this? Start from the names, I'd suggest: `avilableSnapModes`, `snapModeEntries`, `availableSnapModeEntries`.